mmcv/utils/logging.py

```diff
@@ -37,6 +37,17 @@ def get_logger(name, log_file=None, log_level=logging.INFO, file_mode='w'):
         if name.startswith(logger_name):
             return logger
 
+    # handle duplicate logs to the console
+    # Starting in 1.8.0, PyTorch DDP attaches a StreamHandler <stderr> (NOTSET)
+    # to the root logger. As logger.propagate is True by default, this root
+    # level handler causes logging messages from rank>0 processes to
+    # unexpectedly show up on the console, creating much unwanted clutter.
+    # To fix this issue, we set the root logger's StreamHandler, if any, to log
+    # at the ERROR level.
+    for handler in logger.root.handlers:
+        if type(handler) is logging.StreamHandler:
```

Comment:
Should we change `if type(handler) is logging.StreamHandler ` to `if isinstance(hanlder, logging.StreamHandler)`? 
please refer to https://stackoverflow.com/questions/1549801/what-are-the-differences-between-type-and-isinstance

Reply:
Actually I tried that first before switching to the current version. Problem with `isinstance` is that it also returns True for subclasses of `logging.StreamHandler`, which unfortunately includes `pytest.logging.LogCaptureHandler` (see here https://docs.pytest.org/en/6.2.x/_modules/_pytest/logging.html for details), and causes unit tests that rely on `caplog` to fail.

Reply:
got it